Lib/test/test_compile.py

```diff
@@ -1409,6 +1409,16 @@ def f():
         for kw in ("except", "except*"):
             exec(code % kw, g, l);
 
+    def test_regression_gh_120225(self):
+        async def name_4():
+            match b'':
+                case True:
+                    pass
+                case name_5 if f'e':
+                    {name_3: name_4 async for name_2 in name_5}
```

Comment:
I guess the exception handling block is coming from inlining this comprehension?